DOCS-763: More improvements to Python code sample in extend (#1622)
viamrobotics/docs: docs/extend/modular-resources/create/_index.md

from viam.components.base import Base
from viam.components.motor import Motor

> from viam.module.module import Module

Author: ```suggestion
from viam.module.types import Reconfigurable
from viam.module.module import Module
```

> from viam.module.types import Reconfigurable

Author: ```
```

> from viam.module.types import Reconfigurable

Author: ```suggestion
```

    """
    MyBase implements a base that only supports set_power (basic forward/back/turn controls) is_moving (check if in motion), and stop (stop all motion).

    It inherits from the built-in resource subtype Base, and has the constructor function ``MyBase.new_base``
    """

    # Here is where we define our new model's colon-delimited-triplet (acme:demo:mybase)
    # acme = namespace, demo = family, mybase = model name.
    MODEL: ClassVar[Model] = Model(ModelFamily("acme", "demo"), "mybase")

>     def __init__(self, name:str, left: str, right: str):

Author: ```suggestion
    def __init__(self, name: str, left: str, right: str):
```

    """
    MyBase implements a base that only supports set_power (basic forward/back/turn controls) is_moving (check if in motion), and stop (stop all motion).

>     It inherits from the built-in resource subtype Base, and has the constructor function ``MyBase.new_base``

Author: ```suggestion
    It inherits from the built-in resource subtype Base and conforms to the ``Reconfigurable`` protocol, which signifies that this component can be reconfigured.
    Additionally, it specifies a constructor function ``MyBase.new_base`` which confirms to the ``resource.types.ResourceCreator`` type required for all models.
```

    # move_straight: unimplemented
    async def move_straight(self, distance: int, velocity: float, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):

>         pass

Author: ```suggestion
        raise NotImplementedError
```

    # spin: unimplemented
    async def spin(self, angle: float, velocity: float, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):

>         pass

Author: ```suggestion
        raise NotImplementedError
```

    # set_velocity: unimplemented
    async def set_velocity(self, linear: Vector3, angular: Vector3, *, extra: Optional[Dict[str, Any]] = None, timeout: Optional[float] = None, **kwargs):

>         pass

Author: ```suggestion
        raise NotImplementedError
```

from viam.components.base import Base
from viam.module.module import Module

> from my_base import MyBase

Author: This could vary obv so idk

async def main():
    """This function creates and starts a new module, after adding all desired resource models.
    Resource creators must be registered to the resource registry before the module adds the resource model. 
    """

>     Registry.register_resource_creator(Base.SUBTYPE, MyBase.MODEL, ResourceCreatorRegistration(MyBase.new, MyBase.validate_config))

Author: ```suggestion
    Registry.register_resource_creator(Base.SUBTYPE, MyBase.MODEL, ResourceCreatorRegistration(MyBase.new_base, MyBase.validate_config))
```

#!/bin/sh
cd `dirname $0`

# Be sure to use `exec` so that termination signals reach the python process,
# or handle forwarding termination signals manually

> exec $PYTHON src/main.py $@

Author: I wanted to split up the examples so that there would be one that would block less on requirements/env setup I think! If you don't think that's a good idea, happy to remove

> exec $PYTHON src/main.py $@

Reviewer: I don't think it'd be a good idea, if the user has multiple python modules it's going to be really hard to keep track of requirements and make sure the versions are right/don't conflict.

> exec $PYTHON src/main.py $@

Author: Ok sounds good! Removed the less complex version!

```

> Make sure that you set up a Python virtual environment in the directory your module is in to compile your resource properly at execution if using the above script.

Reviewer 2: Does this just require `pip3 install --target=folder_name viam-sdk` - if so might be easier to add that here. If not ignore me

> Make sure that you set up a Python virtual environment in the directory your module is in at this script's execution to compile your resource properly at runtime if using the above script.

Author: ```suggestion
Make sure that you set up a Python virtual environment in the directory your module is in to compile your resource properly at execution if using the above script.
```
